From 88359aa81c4b71389055020665b7c4bdeb3aed50 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Thu, 21 May 2015 10:25:49 -0700
Subject: [PATCH] sdhci: Add checks around hw_mutex_is_locked()

Macro EMMC_HW_MUTEX_IS_LOCKED() calls hw_mutex_is_locked(). Add a call
to netss_driver_ready() to be sure that the HW Mutex is ready.
---
 include/linux/mmc/sdhci.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/mmc/sdhci.h b/include/linux/mmc/sdhci.h
index 919fdb3..58ed379 100644
--- a/include/linux/mmc/sdhci.h
+++ b/include/linux/mmc/sdhci.h
@@ -210,7 +210,8 @@ static void unlock_emmc_access(void)
   mutex_unlock(&mmc_access_lock);
 }
 
-#define EMMC_HW_MUTEX_IS_LOCKED(host) (hw_mutex_is_locked(HW_MUTEX_EMMC))
+#define EMMC_HW_MUTEX_IS_LOCKED(host) (netss_driver_ready() &&\
+  hw_mutex_is_locked(HW_MUTEX_EMMC))
 
 #define LOCK_EMMC_HW_MUTEX(host) do{\
   if (((struct sdhci_host *)host->private)->flags & SDHCI_SUPPORT_HW_MUTEX)\
-- 
1.7.9.5

